proposed/http-client/http-client-meta.md

```diff
@@ -23,24 +23,23 @@ clients will allow libraries to be decoupled from specific implementation such a
 
 ### Non-Goals
 
-* Support for asynchronous HTTP requests is left for another future PSR;
+* Support for asynchronous HTTP requests is left for another future PSR.
 * This PSR does not define how to configure an HTTP client. It only specifies the
-  default behaviours;
-* This PSR is neutral about the use of middlewares (PSR-15).
+  default behaviours.
+* This PSR is neutral about the use of middleware.
 
 #### Asynchronous HTTP client
 
 The reason asynchronous requests are not covered by this PSR is the lack of a
-common standard for Promises. And an HTTP client PSR should not define its own
-promises. At the time the HTTP client PSR was written there was no final PSR
-for Promises.
+common standard for Promises. Promises are sufficiently complex enough that they
+deserve their own specification, and should not be wrapped into this one.
 
 A separate interface for asynchronous requests can be defined in a separate PSR
-once the Promise PSR is accepted. The method signature for asynchronous requests
-has to be different from the method signature for synchronous requests because
-the return type of asynchronous calls will be a Promise. Thus this PSR is forward
-compatible and clients will be able to implement one or both interfaces as makes
-sense for them.
+once a Promise PSR is accepted. The method signature for asynchronous requests
+MUST be different from the method signature for synchronous requests because
+the return type of asynchronous calls will be a Promise. Thus this PSR is forwards
+compatible, and clients will be able to implement one or both interfaces based
+on the features they wish to expose.
```

Comment:
I rephrased this section a ton. I'm still not sure it reads well, but I think it reads better than before.